odoo/addons/base/models/ir_http.py

```diff
@@ -399,7 +403,7 @@ def _binary_record_content(
         _, existing_extension = os.path.splitext(filename)
         if not existing_extension or default_filename:
             extension = mimetypes.guess_extension(mimetype)
-            if extension:
+            if extension and not filename.endswith(extension):
```

Comment:
IMO the only fix that was needed originally was to better detect the presence of an existing extension. E.g. always add the extension if the filename does not already end with a valid extension. Worst case there will be 2 extensions, not a big deal.
Might need a small exclusion list for dual extensions (e.g. jpg vs jpeg, you want to avoid .jpg.jpeg). But that's better than custom model-dependent logic.

Reply:
As such ?

https://github.com/odoo/odoo/pull/88134/files#diff-e1e09396e76ebba8e6de18eeb94d1360adc74f49be53e0cb5b17bf5e6212f6e3R121-R123